### PR TITLE
fix Received RecordBatch prior to Schema for query

### PR DESCRIPTION
**Pull request by RoyZhang2022**
Fix error "Received RecordBatch prior to Schema" at client side when a query is executed.

### CODE REVIEW

File: src/odbc_command_handler.rs [hunk 1]
```diff
@@ -73,15 +72,17 @@ impl OdbcCommandHandler {
...
+        self.send_schema_from_cursor(req.response_sender, &mut cursor, ticket)
     }
 
     fn handle_get_command_data(&mut self, req: GetCommandDataRequest) -> Result<(), MyServerError> {
         let connection = self.get_connection()?;
-        let cursor = self.get_result_cursor(&connection, req.command.clone())?;
-        self.send_flight_data_from_cursor(req.response_sender, cursor)
+        let mut cursor = self.get_result_cursor(&connection, req.command.clone())?;
+        let schema = arrow_odbc::arrow_schema_from(&mut cursor)?;
+        let unmut_cursor = cursor;
```
Comment (jonathanl-bq) [inline src/odbc_command_handler.rs:84]: Why do you need to rebind cursor?

File: src/util.rs [hunk 2]
```diff
@@ -144,3 +162,31 @@ impl<'a> Deref for SchemaAsIpc<'a> {
...
 }
+
+// Convert `RecordBatch`es to wire protocol `FlightData`s
+pub fn batches_to_flight_data(
+    schema: &Schema,
+    batches: Vec<RecordBatch>,
+) -> Result<Vec<FlightData>, ArrowError>
+{
+    let options = IpcWriteOptions::default();
+    let schema_flight_data: FlightData = crate::util::SchemaAsIpc::new(schema, &options).into();
```
Comment (jonathanl-bq) [inline src/util.rs]: nit: Why not just add SchemaAsIpc to the existing `use crate::util`?

EDIT: I didn't realize I was looking at a different file. You could still just add this to the existing `use crate`

EDIT 2: Wait, this is the util module. You should just be using `SchemaAsIpc::new` without the `crate::util`.

File: src/odbc_command_handler.rs (hunk 1 above)
Comment (RoyZhang2022) [reply src/odbc_command_handler.rs:84]: To get a schema, it needs a mutable cursor. But the following function call needs a unmutable cursor.

File: src/util.rs (hunk 2 above)
Comment (RoyZhang2022) [reply src/util.rs]: Done. Removed `crate::util`

**Pull request merged**